egs/wsj/s5/steps/dict/prons_to_lexicon.py

```diff
@@ -0,0 +1,191 @@
+            "set-max-to-one and set-sum-to-one as true or false.")
+
+    return args
+
+def ReadStats(args):
+    lexicon = {}
+    word_count = {}
+    for line in args.stats_file_handle:
+        splits = line.strip().split()
+        if len(splits) < 3:
+            continue
+
+        word = splits[1]
+        count = float(splits[0])
+        phones = ' '.join(splits[2:])
+
+        lexicon[(word, phones)] = lexicon.get((word, phones), 0) + count
+        word_count[word] = word_count.get(word, 0) + count
+
+    return [lexicon, word_count]
+
+def ReadLexicon(lexicon_file_handle):
+    lexicon = set()
+    if lexicon_file_handle:
+        for line in lexicon_file_handle.readlines():
+            splits = line.strip().split()
+            if len(splits) == 0:
+                continue
+            if len(splits) < 2:
+                raise Exception('Invalid format of line ' + line
+                                    + ' in lexicon file.')
+            word = splits[0]
+            try:
+                phones = ' '.join(splits[2:])
+            except ValueError:
+                phones = ' '.join(splits[1:])
+            lexicon.add((word, phones))
+    return lexicon
+
+def ConvertWordCountsToProbs(args, lexicon, word_count):
+    word_probs = {}
+    for entry, count in lexicon.iteritems():
+        word = entry[0]
+        phones = entry[1]
+        prob = float(count) / float(word_count[word])
+        if word in word_probs:
+            word_probs[word].append((phones, prob))
+        else:
+            word_probs[word] = [(phones, prob)]
+
+    return word_probs
+
+def ConvertWordProbsToLexicon(word_probs):
+    lexicon = {}
+    for word, entry in word_probs.iteritems():
+        for x in entry:
+            lexicon[(word, x[0])] = lexicon.get((word,x[0]), 0) + x[1]
+    return lexicon
+
+def NormalizeLexicon(lexicon, set_max_to_one = True,
+                     set_sum_to_one = False, min_prob = 0):
+    word_probs = {}
+    for entry, prob in lexicon.iteritems():
+        t = word_probs.get(entry[0], (0,0))
+        word_probs[entry[0]] = (t[0] + prob, max(t[1], prob))
+
+    for entry, prob in lexicon.iteritems():
+        if set_max_to_one:
+            prob = prob / word_probs[entry[0]][1]
+        elif set_sum_to_one:
+            prob = prob / word_probs[entry[0]][0]
+        if prob < min_prob:
+            prob = 0
+        lexicon[entry] = prob
+
+def WriteLexicon(args, lexicon, filter_lexicon):
+    words = set()
+    num_removed = 0
+    num_filtered = 0
+    for entry, prob in lexicon.iteritems():
+        if prob == 0:
+            num_removed += 1
+            continue
+        if entry in filter_lexicon:
+            num_filtered += 1
+            continue
+        words.add(entry[0])
+        print("{0} {1}".format(entry[0], entry[1]),
+                file = args.out_lexicon_handle)
+    print ("Before pruning, the total num. pronunciations is: {}".format(len(lexicon)))
```

Comment:
I think it's best to print things like this to stderr, as a general rule.  That way you can redirect the output to /dev/stdout and things will work.